### PR TITLE
API router

### LINKED ISSUE #37: Add API Router
- [x] Create an `api` directory
- [x] Add an `index.ts` file to the directory
- [x] Create a router inside the api index
- [x] Add the router to your Express app middleware

### PR DESCRIPTION
**Pull request by dyazdani**
## Description
Setting up router for /api

## Related Issue
Closes #37

## Tests
Screen Shot 2023-09-27 at 12 06 50
Screen Shot 2023-09-27 at 12 05 52
Screen Shot 2023-09-27 at 12 13 16
Screen Shot 2023-09-27 at 12 02 37

### CODE REVIEW

File: server/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import express from 'express';
+
+const apiRouter = express.Router();
+
+//TODO: Fix issue that is making localhost:3000/api return with error saying "Cannot GET /api"
```
Comment (paloobi) [inline server/api/index.ts]: I'm not encountering this issue anymore?

File: server/api/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    try {
+        res.send('API is live');
+    } catch (e) {
+        next(e);
+    }
+})
+
+apiRouter.use((req, res): void => {
+    res.status(404)
+    .send({message: 'Invalid API endpoint'});
```
Comment (paloobi) [inline server/api/index.ts]: nit: when chaining methods, I personally like when the second line is indented once to indicate it part of the expression above

```suggestion
    res.status(404)
        .send({message: 'Invalid API endpoint'});
```

**Review by paloobi — CHANGES_REQUESTED** [review]
PR description should include:

- manually testing /api (screenshot of browser)
- manually testing API 404 (screenshot of browser)
- manually testing 500 (screenshot of browser)

File: server/api/index.ts (hunk 1 above)
Comment (dyazdani) [reply server/api/index.ts]: We thought we deleted that TODO

**Comment by dyazdani** [comment]
@paloobi Just uploaded pictures in the PR description of testing using Postman.

**Review by paloobi — APPROVED** [review]

**Pull request merged**